src/pybind/cephfs.py

```diff
@@ -82,6 +83,12 @@ class cephfs_statvfs(Structure):
                 ("f_flag", c_uint),
                 ("f_namemax", c_uint)]
 
+class cephfs_dirent(Structure):
+    _fields_ = [("d_ino", c_long),
+                ("d_off", c_ulong),
+                ("d_reclen", c_ushort),
+                ("d_type", c_uint8),
+                ("d_name", c_char*256)]
```

Comment:
It's not clear to me that this c_char*256 definition is safe.  dirents are variable-length structures (http://www.delorie.com/gnu/docs/dirent/dirent.5.html), so this would be safe if it was only reading up to a NULL in the string.

That might be safe *if* we explicitly truncate the array we return from the python readdir() wrapper.  I'm guessing that this currently works because the caller is always just doing str() on it and consequently stopping at the null anyway, but we shouldn't rely on that (e.g. if caller did a deepcopy() on the struct that was returned it could segfault).

However, it's also possible that I'm missing something in the ctypes definitions that makes this safe, so please shout if that's the case!